src/main/java/org/moqucu/games/nightstalker/controller/GameLoop.java

```diff
@@ -66,6 +70,23 @@ public void handle(long currentNanoTime) {
                             approachableSprite.approachedBy(approachingSprites);
                         }
                 );
+
```

Comment:
This is procedural style: the gameloop basically is in control of the object's actions. In this case, it will decide on behalf of the robot when to fire. A different approach would be to delegate this behavior to the robot: if the robot spots the night stalker in the line of fire, it will shoot. The latter approach would be more object-oriented. That said the currently chosen approach works.